src/main/java/com/accounts/storage/AccountsRepository.java

```diff
@@ -0,0 +1,36 @@
+package com.accounts.storage;
+
+import com.accounts.model.Account;
+
+import java.math.BigInteger;
+import java.util.Collection;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class AccountsRepository {
+
+    private Map<String, Account> accounts = new ConcurrentHashMap<>();
+
+    public AccountsRepository() {
+    }
+
+    public Collection<Account> getAll() {
+        return accounts.values();
+    }
+
+    public Account get(String id) {
+        Account account = accounts.get(id);
+        if (account == null) {
+            throw new IncorrectRequstedAccountIdException("Account " + id + " does not exist.");
+        } else {
+            return account;
+        }
+    }
+
+    public void add(String id, BigInteger amount) {
+        Account old = accounts.putIfAbsent(id, new Account(id, amount));
+        if (old != null) {
```

Comment:
Shouldn't it be by equals?
account.equals(null) ?

Reply:
To compare with null that way is better. It prevents from null pointer exception